src/AbstractAlgebra.jl

```diff
@@ -1238,13 +1238,26 @@ export Generic
 
 ###############################################################################
 #
-#   Polynomial Ring S, x = R["x"] syntax
+#   misc
+#
+###############################################################################
+
+include("misc/ProductIterator.jl")
+include("misc/Evaluate.jl")
+include("misc/VarNames.jl")
+
+###############################################################################
+#
+#   Polynomial Ring S, x = R[:x] syntax
 #
 ###############################################################################
 
 getindex(R::NCRing, s::VarName) = polynomial_ring(R, s)
-getindex(R::NCRing, s::VarName, ss::VarName...) =
-   polynomial_ring(R, [Symbol(x) for x in (s, ss...)])
+# For compatibility `R[:x, :y]` returns `S, [x, y]` instead of `S, x, y`
+getindex(R::Ring, s1::VarNames, s::VarNames...) = ((S, x...) = polynomial_ring(R, s1, s...); (S, collect(x)))
+getindex(R::Ring, s::Tuple{Vararg{VarNames}}) = ((S, x...) = polynomial_ring(R, s); (S, collect(x)))
```

Comment:
I guess this one also could use a comment explaining it (because I don't get it)... Would this be accurate?
```suggestion
# `R[(:x, :y]` returns `S, [x, y]` instead of `S, x, y`
getindex(R::Ring, s::Tuple{Vararg{VarNames}}) = ((S, x...) = polynomial_ring(R, s); (S, collect(x)))
```
If so, why do we offer this method in the first place?
If not, well, please add a correct comment :-)

Reply:
Mhm, right, we don’t need it. It allows to write `R[names]`, or `R[()]`, or `R[(:x,)]`. But if you need that, you should better use `polynomial_ring`.